### PR TITLE
Throws MSALClientException when back is pressed

### PR DESCRIPTION
**Pull request by SameerK-MSFT**
Fixes 2991 .

**Changes proposed in this request**
URL parameters are analyzed to check if it is user cancelled and corresponding result is returned.

**Testing**
Unit Test

**Performance impact**
None

### CODE REVIEW

File: src/client/Microsoft.Identity.Client/UI/AuthorizationResult.cs [hunk 1]
```diff
@@ -73,6 +73,11 @@ private static AuthorizationResult FromParsedValues(Dictionary<string, string> p
         {
             if (parameters.ContainsKey(TokenResponseClaim.Error))
             {
+                if (parameters.ContainsKey(TokenResponseClaim.ErrorSubcode) && parameters[TokenResponseClaim.ErrorSubcode].Equals(TokenResponseClaim.ErrorSubcodeCancel, StringComparison.OrdinalIgnoreCase))
```
Comment (bgavrilMS) [inline src/client/Microsoft.Identity.Client/UI/AuthorizationResult.cs:76]: `if (parameters.TryGetValue(key, out val)`

**Review by bgavrilMS — APPROVED** [review]
Small suggestion

File: src/client/Microsoft.Identity.Client/UI/AuthorizationResult.cs (hunk 1 above)
Comment (SameerK-MSFT) [reply src/client/Microsoft.Identity.Client/UI/AuthorizationResult.cs]: Good catch. Fixed.

**Review by bgavrilMS — APPROVED** [review]

**Review by jmprieur — APPROVED** [review]
LGTM
Thanks @SameerK-MSFT

**Pull request merged**